### PR TITLE
Adding GitHub checks

### PR DESCRIPTION
**Pull request by gitstart**
CLoses out issue: [FLOW-GS-2](https://developers.gitstart.com/client/flow/ticket/FLOW-GS-2)

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @richardguerre   this pr is ready for review.

NB: The backend tests is failing and stuck in a loop . You can  create a separate ticket for that  if you would like us to fix that.

**Review by richardguerre — CHANGES_REQUESTED** [review]
This is a good start and I will merge it in if you just fix the server test running in an infinite loop issue.

**Comment by richardguerre** [comment]
Also why was this not done as part of https://github.com/richardguerre/flow/pull/2? Can you try not to create duplicate PRs?

File: .github/actions/setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: "Setup"
+description: "Sets up environment"
+inputs:
+  ssh-private-key:
+    description: "SSH key used for private git clone"
+    required: true
```
[the remark this answers is not in the document]
Comment (udochukwu) [reply .github/actions/setup/action.yml:6]: Yeah. It's not needed. We'll remove it.

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+jobs:
+  format-check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Setup environment (node, npm install)
+        uses: ./.github/actions/setup
```
[the remark this answers is not in the document]
Comment (udochukwu) [reply .github/workflows/checks.yml:16]: Yes. It's required in the 2 files, else the flow wont run.

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    env:
+      DATABASE_URL: postgresql://postgres@localhost:5432/flow
+
```
Comment (richardguerre) [inline .github/workflows/checks.yml:50]:
```suggestion
    env:
      DATABASE_URL: postgresql://postgres@localhost:5432/flow
      CI: true

```

File: .github/actions/setup/action.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Check out repository code
+      uses: actions/checkout@v3
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: "18.13.0"
```
Comment (richardguerre) [inline .github/actions/setup/action.yml:15]:
```suggestion
        node-version: "18.12.1"
```

File: .github/actions/setup/action.yml (hunk 1 above)
Comment (richardguerre) [inline .github/actions/setup/action.yml:6]:
```suggestion
description: "Sets up environment"
```

File: .github/workflows/checks.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    env:
+      DATABASE_URL: postgresql://postgres@localhost:5432/flow
+      CI: true
```
Comment (richardguerre) [inline .github/workflows/checks.yml:50]:
```suggestion
    env:
      CI: true
      DATABASE_URL: postgresql://postgres@localhost:5432/flow
      ORIGIN: http://localhost:4000
```

File: .github/workflows/checks.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Setup environment (node, npm install)
+        uses: ./.github/actions/setup
+      - name: Prisma Gen
+        run: npm run gen -w apps/server
+      - name: Test check
+        run: npm run test -w apps/server
```
Comment (richardguerre) [inline .github/workflows/checks.yml:62]:
```suggestion
        run: npm run test -w apps/server -- test/utils/pluralize.test.ts
```

File: .github/workflows/checks.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      DATABASE_URL: postgresql://postgres@localhost:5432/flow
+      ORIGIN: http://localhost:4000
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Setup environment (node, npm install)
+        uses: ./.github/actions/setup
+      - name: Prisma Gen
+        run: npm run gen -w apps/server
```
Comment (richardguerre) [inline .github/workflows/checks.yml:60]:
```suggestion
      - name: Run DB migrations
        run: npm run db:dev
```

File: .github/workflows/checks.yml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Test check
+        run: npm run test -w apps/server -- test/utils/pluralize.test.ts
```
Comment (richardguerre) [inline .github/workflows/checks.yml:62]:
```suggestion
      - name: Test check
        run: npm run test -w apps/server -- test/graphql/Item.test.ts
```

File: .github/workflows/checks.yml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      DATABASE_URL: postgresql://postgres@localhost:5432/flow
+      ORIGIN: http://localhost:4000
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Setup environment (node, npm install)
+        uses: ./.github/actions/setup
+      - name: Run DB migrations
+        run: npm run db:dev
```
Comment (richardguerre) [inline .github/workflows/checks.yml:60]:
```suggestion
        run: DATABASE_URL=postgres://postgres:postgres@localhost:5432/flow_test npm run db:dev
```

File: .github/workflows/checks.yml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    env:
+      CI: true
+      NODE_ENV: test
+      DATABASE_URL: postgresql://postgres@localhost:5432/flow
```
Comment (richardguerre) [inline .github/workflows/checks.yml:51]:
```suggestion
      DATABASE_URL: postgresql://postgres@localhost:5432/flow # `flow` will get replaced with `flow_test` (see .vitest/prisma.ts) as `flow` is the local development DB and `flow_test` is the test DB.
```

File: .github/workflows/checks.yml [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        uses: ./.github/actions/setup
+      - name: Build check
+        run: npm run build
+
+  server-test:
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: pg
```
Comment (richardguerre) [inline .github/workflows/checks.yml:36]:
```suggestion
          POSTGRES_PASSWORD: postgres
```

File: .github/workflows/checks.yml [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Setup environment (node, npm install)
+        uses: ./.github/actions/setup
+      - name: Run DB migrations
+        run: DATABASE_URL=postgres://postgres:postgres@localhost:5432/flow_test npm run db:dev
+      - name: Test check
+        run: npm run test -w apps/server -- test/graphql/Item.test.ts
```
Comment (richardguerre) [inline .github/workflows/checks.yml:62]:
```suggestion
  # server-test:
  #   services:
  #     postgres:
  #       image: postgres
  #       env:
  #         POSTGRES_PASSWORD: postgres
  #         POSTGRES_DB: flow_test
  #       options: >-
  #         --health-cmd pg_isready
  #         --health-interval 10s
  #         --health-timeout 5s
  #         --health-retries 5
  #       ports:
  #         - 5432:5432

  #   runs-on: ubuntu-latest
  #   timeout-minutes: 5
  #   env:
  #     CI: true
  #     NODE_ENV: test
  #     DATABASE_URL: postgresql://postgres@localhost:5432/flow # `flow` will get replaced with `flow_test` (see .vitest/prisma.ts) as `flow` is the local development DB and `flow_test` is the test DB.
  #     ORIGIN: http://localhost:4000

  #   steps:
  #     - name: Check out repository code
  #       uses: actions/checkout@v3
  #     - name: Setup environment (node, npm install)
  #       uses: ./.github/actions/setup
  #     - name: Run DB migrations
  #       run: DATABASE_URL=postgres://postgres:postgres@localhost:5432/flow_test npm run db:dev
  #     - name: Test check
  #       run: npm run test -w apps/server -- test/graphql/Item.test.ts
```

File: .github/workflows/checks.yml [hunk 13]
```diff
@@ -0,0 +1,62 @@
+name: CI/CD Checks
```
Comment (richardguerre) [inline .github/workflows/checks.yml:1]:
```suggestion
name: CI
```

File: .github/workflows/checks.yml [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+      ORIGIN: http://localhost:4000
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Setup environment (node, npm install)
+        uses: ./.github/actions/setup
+      - name: Run DB migrations
+        run: DATABASE_URL=postgres://postgres:postgres@localhost:5432/flow_test npm run db:dev
+      - name: Test check
```
Comment (richardguerre) [inline .github/workflows/checks.yml:61]:
```suggestion
      - name: Run tests
```

**Pull request merged**